app/src/org/commcare/models/database/SqlStorage.java

```diff
@@ -471,8 +471,9 @@ public T read(int id) {
         }
 
         try {
-            c.moveToFirst();
-            return c.getBlob(c.getColumnIndexOrThrow(DatabaseHelper.DATA_COL));
+            if (c != null && c.moveToFirst()) {
+                return c.getBlob(c.getColumnIndexOrThrow(DatabaseHelper.DATA_COL));
+            }
         } finally {
```

Comment:
Tests are failing because when `c == null` I don't return anything. Can anyone think of what the right behavior should be here? Maybe define a `InvalidQueryException` that extends `RuntimeException` and throw that? Or just close this PR and ignore the issue ...